aten/src/ATen/native/cuda/UpSampleNearest2d.cu

```diff
@@ -222,44 +333,83 @@ static void upsample_nearest2d_backward_out_cuda_template(
   int input_height = input_size[2];
   int input_width = input_size[3];
 
-  Tensor grad_output = grad_output_.contiguous();
+  const float height_scale = compute_scales_value_backwards<float>(scales_h, output_height, input_height);
+  const float width_scale = compute_scales_value_backwards<float>(scales_w, output_width, input_width);
 
-  if (grad_input.numel() == 0) {
-    return;
-  }
+  auto memory_format = grad_output_.suggest_memory_format();
+
+  if (memory_format == at::MemoryFormat::ChannelsLast) {
+    Tensor grad_output = grad_output_.contiguous(memory_format);
+    grad_input.unsafeGetTensorImpl()->empty_tensor_restride(memory_format);
```

Comment:
you can't just restride if grad_input is for some reason not memory-dense

Reply:
I added some comments here in the code. I'm not familiar with "structured kernels" and here is the problem. `grad_input` is passed in as a `const Tensor`, and it's not "channels-last contiguous".

If as you said, `grad_input` is not memory-dense, the only option we have is to do calculation on another "channels-last contiguous" tensor, and then copy the result back here.

The current contiguous path is also using `grad_input.data_ptr<scalar_t>()` directly without calling a `.contiguous()` first.